atlasdb-conjure/src/test/java/com/palantir/atlasdb/http/v2/RetryOnSocketTimeoutExceptionProxyTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.http.v2;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import org.junit.Test;
+
+public class RetryOnSocketTimeoutExceptionProxyTest {
```

Comment:
Can you add some tests that mock the supplier to fail say 3 times, then succeed, and verify that it passes? For example:

```java
Supplier<Long> supplier = mock(Supplier.class);
when(supplier.get())
    .thenThrow(new RuntimeException(new SocketTimeoutException()))
    .thenThrow(...)
    .thenThrow(...)
    .thenReturn(11L);

```

Reply:
(And some other tests of this form for no failures, only failures, failures where exceptions don't have the right root cause, etc.)